lib/ansible/module_utils/basic.py

```diff
@@ -1518,9 +1530,16 @@ def _log_to_syslog(self, msg):
             syslog.openlog(str(module), 0, facility)
             syslog.syslog(syslog.LOG_INFO, msg)
 
-    def debug(self, msg):
+    def debug(self, msg, pretty_print=False):
         if self._debug:
-            self.log(msg)
+            if self.debug_logger:
+                if pretty_print:
+                    msg_txt = json.dumps(msg, sort_keys=True, indent=4, separators=(',', ': '))
```

Comment:
This could be done in the logging Formatter (and/or a logging.Filter). And then the modules
could just pass objects to self.debug_logger.debug(), and the handler/formatter decides how to pretty print.